packages/celotool/src/lib/geth.ts

```diff
@@ -520,26 +575,67 @@ export const simulateClient = async (
     // randomly choose which gas currency to use
     const feeCurrencyGold = Boolean(Math.round(Math.random()))
 
-    let feeCurrency
-    if (!feeCurrencyGold) {
-      try {
-        feeCurrency = await kit.registry.addressFor(CeloContract.StableToken)
-      } catch (error) {
-        tracerLog({
-          tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
-          error: error.toString(),
-          ...baseLogMessage,
-        })
+    let feeCurrency, txOptions
+    try {
+      feeCurrency = feeCurrencyGold ? '' : await kit.registry.addressFor(CeloContract.StableToken)
+    } catch (error) {
+      tracerLog({
+        tag: LOG_TAG_CONTRACT_ADDRESS_ERROR,
+        error: error.toString(),
+        ...baseLogMessage,
+      })
+    }
+
+    try {
+      feeCurrency = feeCurrency || ''
+      baseLogMessage.feeCurrency = feeCurrency
+
+      const gasPriceMinimum = await kit.contracts.getGasPriceMinimum()
+
+      const gasPriceBase = feeCurrency
+        ? await gasPriceMinimum.getGasPriceMinimum(feeCurrency)
+        : await gasPriceMinimum.gasPriceMinimum()
+      let gasPrice = new BigNumber(gasPriceBase).times(2)
+
+      // Check if last tx was mined. If not, reuse the same nonce
+      if (lastTx === '' || lastNonce === 0) {
```

Comment:
To get the nonce to use, it'd be easier to not use `kit.connection.nonce()`.  As is, we're using that function, which uses "pending" as the block, and then having to adjust based on the cases.  Instead we can use web3's normal `getTransactionCount`, which uses the default block number, which is "latest".  If we want to be extra safe in case the default is changed (though I can't imagine us ever doing that in contractkit), we can even specify "latest":

```
nonce = await kit.connection(kit.defaultAccount, "latest")
```

This should will work for all three cases handled here.